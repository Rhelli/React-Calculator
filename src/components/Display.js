import React from 'react';
import PropTypes from 'prop-types';

const Display = props => {
  const value = { props };
  return (
    <div>
      <h3>{value}</h3>
    </div>
  );
};

Display.propTypes = {
  value: PropTypes.string,
};

Display.defaultProps = {
  value: 'something',
};
