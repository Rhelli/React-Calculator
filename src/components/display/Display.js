import React from 'react';
import PropTypes from 'prop-types';
import styles from './display.module.scss';

const Display = props => {
  const { value } = props;
  return (
    <div className={styles.display}>
      <h2>{value || '0'}</h2>
    </div>
  );
};

Display.defaultProps = {
  value: '0',
};

Display.propTypes = {
  value: PropTypes.string,
};

export default Display;
