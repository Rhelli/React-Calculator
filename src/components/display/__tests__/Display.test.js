import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Display from '../Display';

describe('The Display Appearance', () => {
  it('Matches the Snapshot', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Initializes with a displayed value of 0', () => {
    const { getByText } = render(<Display />);
    getByText('0');
  });
})
