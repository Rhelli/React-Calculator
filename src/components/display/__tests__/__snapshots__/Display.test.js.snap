// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Display Appearance Matches the Snapshot 1`] = `
<div
  className="display"
>
  <h2>
    0
  </h2>
</div>
`;
